sourcecode/chesssimul/board.py

```diff
@@ -222,8 +235,50 @@ def move_bishop(self, move: Move) -> bool:
     def move_pawn(self, move: Move) -> bool:
         start_square: Square = self.board[move.start[0]][move.start[1]]
         end_square: Square = self.board[move.end[0]][move.end[1]]
-        # if move.end[0] == move.start[0] + 2 or move.end[0] == move.start[0] - 2:
-        return False
+        # Pawns can only move forward one square at a time
+        if start_square.piece.color == Color.BLACK:
+            if move.end[0] == move.start[0] + 1 and move.end[1] == move.start[1] and end_square.isEmpty:
+                if move.end[0] == 7:
+                    self.promotion(move)
+                else:
+                    self.move_piece(move)
+            # except for their very first move where they can move forward two squares.
+            elif move.end[0] == move.start[0] + 2 and move.start[0] == 1 and move.end[1] == move.start[
+                1] and end_square.isEmpty:
+                self.setEnPassantTargetSquare(move.end[0] - 1, move.end[1])
+                self.move_piece(move)
+            # Pawns can only capture one square diagonally in front of them
+            elif move.end[0] == move.start[0] + 1 and move.end[1] == move.start[1] - 1 or move.end[1] == move.start[1] + 1:
```

Comment:
I'm not sure this condition is working as intended : you should are parenthesis around the or statement : 
move.end[0] == move.start[0] + 1 and (move.end[1] == move.start[1] - 1 or move.end[1] == move.start[1] + 1):